Source/WebInspectorUI/UserInterface/Views/CookiePopover.js

```diff
@@ -224,6 +237,16 @@ WI.CookiePopover = class CookiePopover extends WI.Popover
 
         this._secureCheckboxElement = createInputRow("secure", WI.unlocalizedString("Secure"), "checkbox", data.secure).inputElement;
 
+        if (data.partitionKey) {
+            let partitionKeyElement = document.createElement("input");
+            partitionKeyElement.type = "text";
+
+            partitionKeyElement.value = data.partitionKey;
+            partitionKeyElement.readOnly = true;
+            this._partitionKeyInputElement = partitionKeyElement;
+        }
+        this._partitionedCheckboxElement = createInputRow("partitioned", WI.unlocalizedString("Partitioned"), "checkbox", data.partitioned, this._partitionKeyInputElement).inputElement;
```

Comment:
rather than have `additionalElement` is there not a way to do this in CSS?

as you have it now the `inputElement` returned by `createInputRow` will be the `<span>` which won't work for the logic in `createRow` or as efficiently with the add `addEventListener`

Reply:
I decided to go back to adding a dedicated row, and I'll improve it in a follow up.